Sources/SWBCore/LibSwiftDriver/LibSwiftDriver.swift

```diff
@@ -177,23 +177,44 @@ public final class SwiftModuleDependencyGraph: SwiftGlobalExplicitDependencyGrap
         }
     }
 
-    public func queryPlanningDependencies(for key: String) throws -> [String] {
+    public func querySwiftmodulesNeedingRegistrationForDebugging(for key: String) throws -> [String] {
         let graph = try registryQueue.blocking_sync {
             guard let driver = registry[key] else {
                 throw StubError.error("Unable to find jobs for key \(key). Be sure to plan the build ahead of fetching results.")
             }
             return driver.intermoduleDependencyGraph
         }
         guard let graph else { return [] }
-        let directDependencies = graph.mainModule.directDependencies ?? []
-        let transitiveDependencies = Set(directDependencies + SWBUtil.transitiveClosure(directDependencies, successors: { moduleID in graph.modules[moduleID]?.directDependencies ?? [] }).0)
+        var swiftmodulePaths: [String] = []
+        swiftmodulePaths.reserveCapacity(graph.modules.values.count)
+        for (_, moduleInfo) in graph.modules.sorted(byKey: { $0.moduleName < $1.moduleName }) {
```

Comment:
Hmm, I think the only exception is that we may still need to explicitly skip over the main module itself here. Its `modulePath` in the graph is just `<Module_Name>.swiftmodule` and that won't be a thing that the linker will be able to resolve.

Reply:
Would also mean that right above this we instead need to:
```
swiftmodulePaths.reserveCapacity(graph.modules.values.count - 1)
```

Reply:
fixed this up in the latest commit - I left the reserveCapacity calls as-is since that's an approximation anyway